rfcs/react-components/no-action-prop-value-standardization.md

```diff
@@ -0,0 +1,168 @@
+
+Currently, the approach we follow for all components is to use a string value like `'off'`, `'none'` or `'default'` for default values of a prop. This happens both on cases where the default prop has and doesn't have an effect on the component.
+
+## Problem statement
+
+There is no standardization for the naming, resulting in our users needing to read documentation to figure out what to use, as the names used might be compared to CSS keywords (like `'unset'`), and requiring specific component knowledge as the user can't reuse this information for other components. It is also misleading to provide a string value that has no actual impact on the component.
+
+## Detailed Design or Proposal
+
+This RFC proposes that we leverage the standard JavaScript default, `undefined`, for attributes instead of a string.
+
+Put simply, when a component has a default state/behavior, it SHOULD HAVE a default value.<br/>
+Example: Card.appearance has `'filled' | 'filled-alternative' | 'outline' | 'subtle'` and is `'filled'` by default.
+
+When a component does not have a default state/behavior, it SHOULD NOT HAVE a default value.<br/>
+Anti-example (current state): Text.font has `'base' | 'monospace' | 'numeric'` and defaults to `'base'`, where `'base'` does not apply any styles<br/>
+Example (proposal): Text.font has `'monospace' | 'numeric'` and has no defaults (i.e. `undefined`)
+
+### Usage
+
+#### With default
+
+```tsx
+// Card.types.ts
+type CardProps = {
+  appearance?: 'filled' | 'filled-alternative' | 'outline' | 'subtle';
+};
+type CardState = {
+  // Required as we need a value for our hooks to work
+  appearance: 'filled' | 'filled-alternative' | 'outline' | 'subtle';
+};
+
+// useCard.ts
+const { appearance = 'filled' /* {...} */ } = props; // Default applied to enforce behavior
+
+const state = {
+  appearance,
+  // {...}
+};
+
+// useCardStyles.ts
+const appearanceLookup = {
+  filled: styles.filled,
+  'filled-alternative': styles.filledAlternative,
+  outline: styles.outline,
+  subtle: styles.subtle,
+} as const;
+
+state.root.className = mergeClasses(
+  cardClassNames.root,
+  styles.root,
+  appearanceLookup[state.appearance],
+  // {...}
+  state.root.className,
+);
+```
+
+#### Without default
+
+```tsx
+// Text.types.ts
+type TextProps = {
+  font?: 'monospace' | 'numeric';
+};
+type TextState = {
+  // Also nullable as the default does not overwrite styles
+  font?: 'monospace' | 'numeric';
+};
+
+// useText.ts
+const { font /* {...} */ } = props; // We no longer set a default here
+
+const state = {
+  font,
+  // {...}
+};
+
+// useTextStyles.ts
+const { font } = state;
+
+const fontLookup = {
+  _default: undefined,
+  base: styles.fontBase,
+  monospace: styles.fontMonospace,
+  numeric: styles.fontNumeric,
+} as const;
+
+state.root.className = mergeClasses(
+  cardClassNames.root,
+  styles.root,
+  fontLookup[font ?? '_default'], // Applying a default to keep usage of a lookup
+  // {...}
+  state.root.className,
+);
```

Comment:
I would also cut this

Reply:
I feel like the implementation pattern should be discussed to avoid individual flavors across the code. Do you think another RFC should tackle it?

Reply:
I dropped this comment before the release as was thinking that the RFC could go before 🐱  We could do changes to avoid breaks in future, but it does not matter as v9 was released. Let's leave implementation details and discuss them 👍

Reply:
I'll go ahead and change implementation to reflect that there was no discussion yet so we can get it merged then